sasmodels/models/octahedron_truncated.c

```diff
@@ -0,0 +1,226 @@
+#include <math.h>
+#include <stdio.h> 
+static double
+form_volume(double length_a, double b2a_ratio, double c2a_ratio, double t)
```

Comment:
Thanks a lot for the contribution! 
I am not really expert on these models but it seems to me there are t values that entire volume can go below zero, which may cause problems along the way.

Reply:
The simple solution to it, may be to contsrain t  parameter in the parameter table.

Reply:
Thank you for explaining. I've been simply looking at this term (1.-3*(1.-t)*(1.-t)*(1.-t)) and where it zeros.